packages/atomic/src/components/recommendations/atomic-recs-interface/atomic-recs-interface.ts

```diff
@@ -0,0 +1,372 @@
+import {
+  VERSION as HEADLESS_VERSION,
+  loadConfigurationActions,
+} from '@coveo/headless';
+import {
+  buildRecommendationEngine,
+  EcommerceDefaultFieldsToInclude,
+  loadFieldActions,
+  loadRecommendationActions,
+  loadSearchConfigurationActions,
+  type RecommendationEngine,
+  type RecommendationEngineConfiguration,
+  type LogLevel as RecsLogLevel,
+} from '@coveo/headless/recommendation';
+import {provide} from '@lit/context';
+import i18next, {type i18n} from 'i18next';
+import {html, LitElement} from 'lit';
+import {customElement, property, state} from 'lit/decorators.js';
+import {bindingsContext} from '@/src/components/common/context/bindings-context';
+import {augmentAnalyticsConfigWithAtomicVersion} from '@/src/components/common/interface/analytics-config';
+import type {CommonBindings} from '@/src/components/common/interface/bindings';
+import {
+  type BaseAtomicInterface,
+  type InitializeEvent,
+  InterfaceController,
+} from '@/src/components/common/interface/interface-controller';
+import {arrayConverter} from '@/src/converters/array-converter';
+import {booleanConverter} from '@/src/converters/boolean-converter';
+import {errorGuard} from '@/src/decorators/error-guard';
+import {watch} from '@/src/decorators/watch';
+import {withTailwindStyles} from '@/src/decorators/with-tailwind-styles';
+import {ChildrenUpdateCompleteMixin} from '@/src/mixins/children-update-complete-mixin';
+import {markParentAsReady} from '@/src/utils/init-queue';
+import {getAnalyticsConfig} from './analytics-config';
+import {createRecsStore, type RecsStore} from './store';
+
+export type RecsInitializationOptions = RecommendationEngineConfiguration;
+export type RecsBindings = CommonBindings<
+  RecommendationEngine,
+  RecsStore,
+  AtomicRecsInterface
+>;
+
+const FirstRecommendationExecutedFlag = 'firstRecommendationExecuted';
+
+const mismatchedInterfaceAndEnginePropError = (
+  interfaceKind: 'search' | 'recommendation',
```

Comment:
Why would the interfaceKind ever be 'search' in the context of an atomic-recs-interface?